### ISSUE TITLE
Feature request: liquid formatting extensions as gems

### ISSUE THREAD

**Issue by haxwithaxe** [huginn/huginn#2006]
Being able to extend the liquid tokens as gems would make a lot of the js agent usage unneeded (at least mine) and make it easier to cleanly reuse formatting tasks.

**Comment by dsander** [comment]
Hi @haxwithaxe,

I think that is already possible. You can use `huginn_agent` to inject any code into your Huginn instance, it does not have to be an Agent. 

```
HuginnAgent.load 'subdirectory/file'
```
Put that command in the auto required file of your "Agent" gem and it the code will be executed. In that file you can create and register new Liquid tags.

**Comment by haxwithaxe** [comment]
Would those tags be available in the core agent models (eg event formatting agent)?

**Comment by dsander** [comment]
Yes, the `liquid` tag registration is global for the whole ruby process.
